### ISSUE TITLE
Adding a nonce to whitelist the output of Rails' javascript_include_tag helper. 

### ISSUE THREAD

**Issue by dqdinh** [twitter/secureheaders#242]
The README suggests calling "... the SecureHeaders.content_security_policy_nonce then set the nonce attributes on the various tags." to create and use a nonce. I'm trying to whitelist the output script from Rails' javascript_include_tag helper:

```<%= javascript_include_tag 'application' , nonce: SecureHeaders.content_security_policy_nonce %>```

but I'm getting this error:

```ActionView::Template::Error (undefined method `content_security_policy_nonce' for SecureHeaders:Module)```

Is there a way to directly call content_security_policy_nonce to generate and attach a nonce to whitelist  Rails' javascript_include_tag helper?

**Comment by oreoshake** [comment]
tl;dr this should work for you:

```erb
<%= javascript_include_tag 'application' , nonce: content_security_policy_script_nonce %>
```

It does appear the docs don't accurately reflect things. `content_security_policy_script_nonce` should work for you (`content_security_policy_nonce` is private).

Also, calling the class method `SecureHeaders.content_security_policy_script_nonce` would require passing in the `request` object where as the helper included with actionview can be called without `SecureHeaders` prefix and without the need to pass the request object.

Lastly, can I ask a question? Nonces are typically only for inline content and are typically not used when sourcing script files. Does it work without the nonce all together? Or does your script src disallow all but nonced sources? If the latter, that's pretty awesome!

**Comment by dqdinh** [comment]
I tried: 
```<%= javascript_include_tag 'application' , nonce: content_security_policy_script_nonce %>```
but running into the same error:
```ActionView::Template::Error (undefined local variable or method `content_security_policy_script_nonce' ```

Do I need to be on 3.x for this to work? I'm currently using version 2.5.2.

With respect to your question: The applicaton.js script will work without the nonce. Secure headers will throw "report only" errors in console:
``[Report Only] Refused to apply inline style because it violates the following Content Security Policy directive: "style-src https: 'self' nonce"". Either the 'unsafe-inline' keyword, a hash ('sha256-XGrHQrUbfkCcNWdkPgEh61nzQllRLXB+PMvrsGIbFNI='), or a nonce ('nonce-...') is required to enable inline execution.``` 

BTW thanks for your prompt response!

**Comment by oreoshake** [comment]
> Do I need to be on 3.x for this to work? I'm currently using version 2.5.2.

Ah yes, that is the problem. Here are the [docs](https://github.com/twitter/secureheaders/blob/2.x/README.md) for the 2.x series. Back then an instance variable `@content_security_policy_nonce` should be available in the view.

**Comment by oreoshake** [comment]
I've added this backwards incompatibility to the [3.x upgrade document](https://github.com/twitter/secureheaders/commit/9739fc4d19718e5c8694d3222b52c91ccceffc9b).

**Comment by dqdinh** [comment]
Thanks!

**Comment by jurre** [comment]
For anyone finding this through search, it looks like (now?) you'll need to use this:

```erb
<%= javascript_include_tag :application, nonce: content_security_policy_nonce(:style) %>
```

See the source: https://github.com/twitter/secureheaders/blob/master/lib/secure_headers/view_helper.rb#L28

I'll send a PR to update the upgrade guide as well as well
